frontend/src/metabase/components/PaginationControls.jsx

```diff
@@ -8,25 +8,37 @@ export default function PaginationControls({
   page,
   pageSize,
   itemsLength,
+  total,
+  showTotal,
   onNextPage,
   onPreviousPage,
 }) {
+  const isPreviousDisabled = page === 0;
+  const isNextDisabled =
+    total != null ? isLastPage(page, pageSize, total) : !onNextPage;
+
   return (
-    <div className="flex align-center" aria-label="pagination">
-      <span className="text-bold mr1">
+    <div className="flex align-center text-bold" aria-label="pagination">
+      <span className="mr1">
         {page * pageSize + 1} - {page * pageSize + itemsLength}
+        {showTotal && (
+          <React.Fragment>
+            <span className="text-light">&nbsp;of&nbsp;</span>
```

Comment:
not the most significant line of text in the app, but I suppose we should localize it

Reply:
prepositions are actually a real pain in the butt for some languages